driver/lchmod_linux.go

```diff
@@ -0,0 +1,19 @@
+package driver
+
+import (
+	"os"
+
+	"github.com/containerd/continuity/sysx"
+)
+
+// Lchmod changes the mode of a file not following symlinks.
+func (d *driver) Lchmod(path string, mode os.FileMode) error {
+	// On Linux, file mode is not supported for symlinks,
+	// and fchmodat() does not support AT_SYMLINK_NOFOLLOW,
+	// so symlinks need to be skipped entirely.
+	if st, err := os.Stat(path); err == nil && st.Mode()&os.ModeSymlink != 0 {
+		return nil
+	}
```

Comment:
This was my original approach, but then I realized I am changing the behavior that way (as previously the error (like ENOENT or EACCESS) was returned from `Fchmodat()`).

This code assumes that if `path` can't be `stat()` it is definitely not a symlink.

Reply:
ahh, because the `os` package will wrap the error?

Reply:
For one thing, yes. I didn't want to inadvertently open a can of worms.

In general it's a problem that is out of scope for this PR. The problem is, some errors returned by continuity come from high-level Go functions (like the ones from `os`) and are of `PathError` type, some come from low-level (like `x/sys/unix` or `syscall`) and are not wrapped. The most user-visible difference is high-level ones contain file names and in general provide more clue about what is going on.

Based on the above, it would be good to review and unify errors returned (I am in favor of `os.PathError{}`). But again it's out of scope for this PR.